File: Tests/AcaiParserGoldenTests/Fixtures/shop.kt
package com.example.shop

interface Priced {
    fun price(): Double
}

abstract class Item(val name: String) : Priced {
    abstract override fun price(): Double
    open fun describe(): String = name
}

enum class Category(val label: String) {
    FOOD("food"),
    TOOL("tool")
}

data class Money(val amount: Double, val currency: String = "EUR")

class Product(name: String, private val base: Double) : Item(name) {

    var discount: Double = 0.0
    private val audit: Logger = Logger()

    class Builder {
        var name: String = ""
        fun build(): Product = Product(name, 0.0)
    }

    companion object {
        fun empty(): Product = Product("", 0.0)
    }

    override fun price(): Double {
        audit.log("pricing")
        return base - discount
    }

    override fun describe(): String {
        val money = Money(price(), "EUR")
        return "$name ${money.amount}"
    }

    fun classify(value: Int): Category {
        if (value > 10) {
            return Category.TOOL
        } else if (value > 5) {
            return Category.FOOD
        }
        for (index in 0 until value) {
            discount += index
        }
        return Category.FOOD
    }
}

class Logger {
    fun log(message: String) {}
}

fun topLevelDiscount(product: Product): Double = product.price()

val DEFAULT_CURRENCY: String = "EUR"

File: Tests/AcaiParserGoldenTests/Fixtures/shop.dart
library shop;

abstract class Priced {
  double price();
}

enum Category { food, tool }

mixin Auditable {
  void audit(String message) {}
}

class Logger {
  void log(String message) {}
}

abstract class Item implements Priced {
  final String name;

  Item(this.name);

  @override
  double price();

  String describe() => name;
}

class Product extends Item with Auditable {
  final Logger _audit = Logger();
  double base;
  double discount = 0;
  static const String defaultCurrency = 'EUR';

  Product(String name, this.base) : super(name);

  factory Product.empty() => Product('', 0);

  @override
  double price() {
    _audit.log('pricing');
    return base - discount;
  }

  @override
  String describe() {
    final local = Logger();
    local.log(name);
    return name;
  }

  Category classify(int value) {
    if (value > 10) {
      return Category.tool;
    } else if (value > 5) {
      return Category.food;
    }
    for (var index = 0; index < value; index++) {
      discount += index;
    }
    return Category.food;
  }

  bool get isFree => base == 0;

  set price2(double value) {
    base = value;
  }

  T identity<T>(T value) => value;
}

extension ProductDescription on Product {
  String shortLabel() => describe();
}

double topLevelDiscount(Product product) => product.price();

const String kDefaultCurrency = 'EUR';
